lib/rucio/core/replica.py

```diff
@@ -3576,23 +4032,47 @@ def get_suspicious_reason(rse_id, scope, name, nattempts=0, logger=logging.log,
     # Alias for bad replicas
     bad_replicas_alias = aliased(models.BadReplicas, name='bad_replicas_alias')
 
-    # query base
-    query = session.query(bad_replicas_alias.scope, bad_replicas_alias.name, bad_replicas_alias.reason, bad_replicas_alias.rse_id)\
-                   .filter(bad_replicas_alias.rse_id == rse_id,
-                           bad_replicas_alias.scope == scope,
-                           bad_replicas_alias.name == name,
-                           bad_replicas_alias.state == 'S',
-                           ~exists(select(1).where(and_(bad_replicas_alias.rse_id == rse_id,
-                                                        bad_replicas_alias.scope == scope,
-                                                        bad_replicas_alias.name == name,
-                                                        bad_replicas_alias.state != 'S',))))
-    count = query.count()
-
-    query_result = query.group_by(bad_replicas_alias.rse_id, bad_replicas_alias.scope, bad_replicas_alias.name, bad_replicas_alias.reason).having(func.count() > nattempts).all()
+    stmt = select(
+        bad_replicas_alias.scope,
+        bad_replicas_alias.name,
+        bad_replicas_alias.reason,
+        bad_replicas_alias.rse_id
+    ).where(
+        and_(bad_replicas_alias.rse_id == rse_id,
+             bad_replicas_alias.scope == scope,
+             bad_replicas_alias.state == 'S',
+             bad_replicas_alias.name == name,
+             ~exists(select(1).where(
+                 and_(bad_replicas_alias.rse_id == rse_id,
+                      bad_replicas_alias.name == name,
+                      bad_replicas_alias.scope == scope,
+                      bad_replicas_alias.state != 'S'))))
+    ).subquery()
+
+    count_query = select(
+        func.count()
+    ).select_from(
+        stmt
+    )
+    count = session.execute(count_query).scalar_one()
+
+    grouped_stmt = select(
+        bad_replicas_alias.scope,
+        bad_replicas_alias.name,
+        bad_replicas_alias.reason,
+        bad_replicas_alias.rse_id
+    ).group_by(
+        bad_replicas_alias.rse_id,
+        bad_replicas_alias.scope,
+        bad_replicas_alias.name,
+        bad_replicas_alias.reason
+    ).having(
+        func.count() > nattempts
+    )
```

Comment:
There is a regression in this query. It does not make use of the existing `query` and, as such, a large portion of it is missing in the new version.